### PR TITLE
Check subtask deadline

### PR DESCRIPTION
**Pull request by Wolf480pl**
Make TaskComputer verify deadline of subtasks by checkig if it's not later than the task's deadline.

This implements #324

### CODE REVIEW

File: golem/task/taskcomputer.py [hunk 1]
```diff
@@ -227,36 +255,44 @@ def run(self):
...
         self.compute_tasks = config_desc.accept_tasks
         self.change_docker_config(config_desc, run_benchmarks, in_background)
-    
+
     def _validate_task_state(self, task_state):
         td = task_state.definition
         if not os.path.exists(td.main_program_file):
-            logger.error("Main program file does not exist: {}".format(td.main_program_file))
+            logger.error("Main program file does not exist: {}"
+                         .format(td.main_program_file))
```
Comment (jiivan) [inline golem/task/taskcomputer.py]: While reformatting please also use `%` substisution.

File: golem/task/taskcomputer.py [hunk 2]
```diff
@@ -384,25 +422,39 @@ def __request_task(self):
...
         task_id = self.assigned_subtasks[subtask_id].task_id
         working_dir = self.assigned_subtasks[subtask_id].working_directory
         unique_str = str(uuid.uuid4())
 
+        task_header = self.task_server.task_keeper.task_headers[task_id]
+        task_deadline = task_header.deadline
+
+        if subtask_deadline > task_deadline:
+            self.subtask_failure(task_id,
+                                 "Subtask deadline is after task deadline: "
```
Comment (jiivan) [inline golem/task/taskcomputer.py]: Maybe something like: `"Subtask deadline reached: "?

File: tests/golem/docker/test_docker_luxrender_task.py [hunk 3]
```diff
@@ -210,9 +215,10 @@ def test_luxrender_real_task_png(self):
         task.res_x = 200
         task.haltspp = 20
         # 1) to make it deterministic,
-        # 2) depending on the kernel, small cropwindow can generate darker img, this is a know issue in lux:
+        # 2) depending on the kernel, small cropwindow can generate darker img,
+        #    this is a know issue in lux:
```
Comment (jiivan) [inline tests/golem/docker/test_docker_luxrender_task.py]: know**n** issue

**Review by jiivan — APPROVED** [review]
Besides my little remarks LGTM :+1:

File: golem/task/taskcomputer.py (hunk 2 above)
Comment (Wolf480pl) [reply golem/task/taskcomputer.py]: No, because it's not reached yet.

Both deadlines are (most likely) in the future.
The thing is, a subtask, which is part of a bigger task, should be finished before the big task is finished.
If the deadlines say eg. "yeah, you still have 10 hours to do this subtask" but the parent task's deadline is in 2 hours, then something is wrong.

File: golem/task/taskcomputer.py (hunk 1 above)
Comment (Wolf480pl) [reply golem/task/taskcomputer.py]: Really? I thought `%` was legacy/deprecated.

Comment (jiivan) [reply golem/task/taskcomputer.py]: It is but it's the only substitution that works with python logging library.
See: https://gist.github.com/jiivan/eedb505574ccb2503af48fa2e3043036

File: golem/task/taskcomputer.py (hunk 2 above)
Comment (jiivan) [reply golem/task/taskcomputer.py]: Oh, now I understand. So maybe: "Subtask deadline shouldn't be after..." So it would be more obvious?

**Comment by Wolf480pl** [comment]
No idea why circle failed :/

**Comment by badb** [comment]
This PR was absolutely fine, there was done what should be done. However after discussion with Grzegorz we decided that it shouldn't be merged now. This will be useful when we add an estimated time that connected with performance may given an estimation if provider is able to compute something or not. 
Without that it may slow down computations for requestor. Please check the issue, for new task description. 
Check the issue for new task description

To sum up: this branch should be closed but not removed - kept for reference with further development process.
